pages/en/lb4/Best-practices-with-Loopback-4.md

```diff
@@ -12,8 +12,10 @@ LoopBack 4 is more than just a framework: It’s an ecosystem that encourages de
 
 Our best practice follows an "API first" and test-driven development approach:
 
-1. [**Defining and validating the API**](./Defining-and-validating-the-API.html): This section guides you through constructing your API first before any internal logic is added.
-2. [**Testing the API**](./Testing-the-API.html): This section describes the process of writing smoke test for your API and its spec.
-3. [**Defining your testing strategy**](./Defining-your-testing-strategy.html): This section discusses the advantages and the process of building a strong testing suite.
-4. [**Implementing features**](./Implementing-features.html): This section demonstrates how the tests for each feature of your application should be written, and how to write the logic to make these tests pass. In the example, the tests for the controller, model, repository, data source, and sequence are written and then implemented.
-5. [**Preparing the API for consumption**](./Preparing-the-API-for-consumption.html): This section shows how the endpoints can be physically tested using the Swagger UI.
+1. **Defining the API**: There are two possible approaches to take in this section
+  - [**Defining the API using bottom-up appraoch**](./Defining-the-API-using-bottom-up-approach.html): This section guides you through setting up a skeleton of your application so that a full API can be automatically generated.
```

Comment:
- Typo in `approach`
- Not sure what you mean by `so that a full API can be automatically generated`; the bottom-up approach is one where you're building the controllers, functions, routes and so on that make up the bulk of your API. The portion that's generated is the OpenAPI/Swagger specification that _describes_ it.